### PR TITLE
Machine ID: Agent based SSH auth for multiplexer

### PR DESCRIPTION
**Pull request by strideynet**
One of the current problems is that TBot does not write to destinations atomically and OpenSSH does not read from destinations atomically - this means that OpenSSH could read during an identity renewal and receive a mismatched certificate and private key. We can solve this problem by offering an agent - with the added benefit that we avoid writing the private key to disk.

Additionally, I tidied up some logging to be less verbose when the connection closes.

### CODE REVIEW

File: lib/config/openssh/openssh.go [hunk 1]
```diff
@@ -266,3 +258,67 @@ func (c *SSHConfig) GetSSHConfig(sb *strings.Builder, config *SSHConfigParameter
...
+	if err != nil {
+		c.log.WithError(err).Debugf("Could not determine SSH version, using default SSH config")
+		sshOptions = getDefaultSSHConfigOptions()
+	} else {
+		c.log.Debugf("Found OpenSSH version %s", version)
+		sshOptions = getSSHConfigOptions(version)
+	}
+	if config.Port == 0 {
+		config.Port = defaults.SSHServerListenPort
+	}
```
[the remark this answers is not in the document]
Comment (strideynet) [reply lib/config/openssh/openssh.go:312]: Yes - unfortunately it looks like we've always done this. I imagine that OpenSSH will refuse to use Port 0 😓

**Review by rosstimothy — APPROVED** [review]

File: lib/config/openssh/openssh.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/config/openssh/openssh.go:312]:
```
Bad port '0'.
```

File: lib/tbot/service_ssh_multiplexer.go [hunk 2]
```diff
@@ -435,21 +453,62 @@ func (s *SSHMultiplexerService) Run(ctx context.Context) (err error) {
...
+				case <-egCtx.Done():
+					return nil
+				case <-time.After(50 * time.Millisecond):
+				}
+
+				continue
+			}
+
+			go func() {
+				defer context.AfterFunc(egCtx, func() { _ = conn.Close() })()
```
Comment (zmb3) [inline lib/tbot/service_ssh_multiplexer.go:492]: Why do we need to defer these `AfterFunc` calls?

**Review by zmb3 — APPROVED** [review]

File: lib/tbot/service_ssh_multiplexer.go (hunk 2 above)
Comment (strideynet) [reply lib/tbot/service_ssh_multiplexer.go:492]: It removes the watcher on the `context.Context` e.g if the root context isn't cancelled and proxying exits for another reason, we don't want to continue waiting for the context to be cancelled.

**Pull request merged**